recipe-client-addon/test/browser/browser_NormandyDriver.js

```diff
@@ -22,10 +23,8 @@ add_task(withDriver(Assert, async function installXpi(driver) {
   is(addonId, "normandydriver@example.com", "Expected test addon was installed");
   isnot(addonId, null, "Addon install was successful");
 
-  // Installing kicks off an asychronous process which tries to read the manifest.json
-  // to startup the addon. Note that onInstallEnded is triggered too early
-  // which is why we need this delay.
-  await new Promise(resolve => SimpleTest.executeSoon(resolve));
+  // Wait until the add-on is fully started up to uninstall it.
+  await AddonTestUtils.promiseWebExtensionStartup("normandydriver@example.com");
```

Comment:
It's best to create the promise before you install/start the extension, so you don't risk adding the startup listener after startup is complete.